cli/lib/kontena/cli/stacks/yaml/service_extender.rb

```diff
@@ -34,12 +34,15 @@ def extend_from(from)
       # @param [Array] to
       # @return [Array]
       def extend_env_vars(from, to)
-        to   ||= []
-        from ||= []
-        to_hash   = Hash[*to.flat_map {|t| t.split('=') }]
-        from_hash = Hash[*from.flat_map {|f| f.split('=') }]
-
-        from_hash.merge(to_hash).map {|k,v| "#{k}=#{v}"}
+        env_vars = to || []
+        if from
+          from.each do |env|
+            env_vars << env unless to && to.find do |key|
```

Comment:
This monster came back.

To handle stuff with only variable, but no value, it could be:
```
  to_hash   = Hash[*to.flat_map {|t| key, value = t.split('=',2); [key, value] }]
```

from the previous version.